src/hypertable_restrict_info.c

```diff
@@ -71,68 +73,108 @@ dimension_restrict_info_create(Dimension *d)
 }
 
 static bool
-dimension_restrict_info_open_add(DimensionRestrictInfoOpen *dri, StrategyNumber strategy, Const *c)
+dimension_restrict_info_open_add(DimensionRestrictInfoOpen *dri, StrategyNumber strategy, List *values, Oid type)
 {
-	int64		value = time_value_to_internal(c->constvalue, c->consttype, false);
+	ListCell   *item;
+	bool		restriction_added;
 
-	switch (strategy)
+	restriction_added = false;
+	foreach(item, values)
 	{
-		case BTLessEqualStrategyNumber:
-		case BTLessStrategyNumber:
-			if (dri->upper_strategy == InvalidStrategy || value < dri->upper_bound)
-			{
-				dri->upper_strategy = strategy;
-				dri->upper_bound = value;
-			}
-			return true;
-		case BTGreaterEqualStrategyNumber:
-		case BTGreaterStrategyNumber:
-			if (dri->lower_strategy == InvalidStrategy || value > dri->lower_bound)
-			{
-				dri->lower_strategy = strategy;
+		Datum		datum = PointerGetDatum(lfirst(item));
+		int64		value = time_value_to_internal(datum, type, true);
+
+		if (value == -1)
+		{
+			elog(WARNING, "unknown time restriction type %d", type);
+			return restriction_added;
+		}
+
+		switch (strategy)
+		{
+			case BTLessEqualStrategyNumber:
+			case BTLessStrategyNumber:
+				if (dri->upper_strategy == InvalidStrategy || value < dri->upper_bound)
+				{
+					dri->upper_strategy = strategy;
+					dri->upper_bound = value;
+				}
+				restriction_added |= true;
```

Comment:
Bitwise OR looks more elegant (at least to me :) but I agree it can bring confusion since we only operate on boolean type so I'll change it. 
The existing code was returning `true` outside a curly braces so I kept the same behaviour. Maybe @cevian can shed some light there?